src/test/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandlerTests.java

```diff
@@ -0,0 +1,232 @@
+        Map<String, Object> mockMapping = new HashMap<>();
+        Map<String, Object> mockMetaMapping = new HashMap<>();
+        mockMetaMapping.put(SCHEMA_VERSION_FIELD, 1);
+        mockMapping.put(META, mockMetaMapping);
+        MappingMetadata mockMappingMetadata = mock(MappingMetadata.class);
+        when(mockIndexMetadata.mapping()).thenReturn(mockMappingMetadata);
+        when(mockMappingMetadata.getSourceAsMap()).thenReturn(mockMapping);
+
+        flowFrameworkIndicesHandler.initFlowFrameworkIndexIfAbsent(index, listener);
+
+        ArgumentCaptor<PutMappingRequest> putMappingRequestArgumentCaptor = ArgumentCaptor.forClass(PutMappingRequest.class);
+        @SuppressWarnings({ "unchecked" })
+        ArgumentCaptor<ActionListener<AcknowledgedResponse>> listenerCaptor = ArgumentCaptor.forClass(ActionListener.class);
+        verify(indicesAdminClient, times(1)).putMapping(putMappingRequestArgumentCaptor.capture(), listenerCaptor.capture());
+        PutMappingRequest capturedRequest = putMappingRequestArgumentCaptor.getValue();
+        assertEquals(index.getIndexName(), capturedRequest.indices()[0]);
+    }
+
+    public void testInitIndexIfAbsent_IndexExist_returnFalse() {
+        FlowFrameworkIndex index = FlowFrameworkIndex.GLOBAL_CONTEXT;
+        indexMappingUpdated.put(index.getIndexName(), new AtomicBoolean(false));
+
+        ClusterState mockClusterState = mock(ClusterState.class);
+        Metadata mockMetadata = mock(Metadata.class);
+        when(clusterService.state()).thenReturn(mockClusterState);
+        when(mockClusterState.metadata()).thenReturn(mockMetadata);
+        when(mockMetadata.hasIndex(index.getIndexName())).thenReturn(true);
+
+        @SuppressWarnings("unchecked")
+        ActionListener<Boolean> listener = mock(ActionListener.class);
+        @SuppressWarnings("unchecked")
+        Map<String, IndexMetadata> mockIndices = mock(Map.class);
+        when(mockClusterState.getMetadata()).thenReturn(mockMetadata);
+        when(mockMetadata.indices()).thenReturn(mockIndices);
+        when(mockIndices.get(anyString())).thenReturn(null);
+
+        flowFrameworkIndicesHandler.initFlowFrameworkIndexIfAbsent(index, listener);
+        assertTrue(indexMappingUpdated.get(index.getIndexName()).get());
+    }
+
+    public void testDoesIndexExist() {
+        ClusterState mockClusterState = mock(ClusterState.class);
+        Metadata mockMetaData = mock(Metadata.class);
+        when(clusterService.state()).thenReturn(mockClusterState);
+        when(mockClusterState.metadata()).thenReturn(mockMetaData);
+
+        flowFrameworkIndicesHandler.doesIndexExist(GLOBAL_CONTEXT_INDEX);
+
+        ArgumentCaptor<String> indexExistsCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockMetaData, times(1)).hasIndex(indexExistsCaptor.capture());
+
+        assertEquals(GLOBAL_CONTEXT_INDEX, indexExistsCaptor.getValue());
+    }
+    */
```

Comment:
Why commenting the tests here?

Reply:
I was facing some issue changing some of these tests while refactoring the code, I now fixed this.